### PR TITLE
Log picked up files in verbose mode

### LINKED ISSUE #268: Add flag for printing picked up files
When developing a bit more complex KPOps projects, we often have a lot of files contributing to a deployment. It would be a nice addition to have the option to let KPOps print all the files that were picked up and contribute to the current command, for example, something like this:

```
> kpops generate pipeline/pipeline.yaml --debug
Picked up: ./pipeline/pipeline.yaml
Picked up: ./pipeline/__init__.py
Picked up: ./config.yaml
Picked up: ./defaults/defaults.yaml
Picked up: ./defaults/defaults_dev.yaml
...
```

### PR DESCRIPTION
**Pull request by raminqaf**
Closes #268 
Depends on #438 
The output looks like this when running KPOps with the `--verbose` command:
```console
Yaml - Picked up: config.yaml
Registry - Picked up: tests/pipeline/test_components/__init__.py
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/pipeline.yaml
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/defaults_development.yaml
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/defaults.yaml
BaseDefaultsComponent - Found defaults for component type streams-app in tests/pipeline/resources/pipeline-with-env-defaults/defaults.yaml
Yaml - Picked up: tests/pipeline/resources/defaults.yaml
BaseDefaultsComponent - Found defaults for component type streams-app in tests/pipeline/resources/defaults.yaml
Yaml - Picked up: tests/pipeline/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kafka-app in tests/pipeline/resources/pipeline-with-env-defaults/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kafka-app in tests/pipeline/resources/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in tests/pipeline/resources/pipeline-with-env-defaults/defaults_development.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in tests/pipeline/resources/pipeline-with-env-defaults/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in tests/pipeline/resources/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in tests/pipeline/defaults.yaml
BaseDefaultsComponent - Found defaults for component type pipeline-component in tests/pipeline/resources/defaults.yaml
BaseDefaultsComponent - Enriching component of type streams-app
[...]
```

### CODE REVIEW

File: kpops/components/base_components/base_defaults_component.py [hunk 1]
```diff
@@ -122,25 +122,23 @@ def extend_with_defaults(cls, config: KpopsConfig, **kwargs: Any) -> dict[str, A
...
-            )
-            + typer.style(cls.type, fg=typer.colors.GREEN, bold=True, underline=True)
-        )
         main_default_file_path, environment_default_file_path = get_defaults_file_paths(
             config, ENV.get("environment")
         )
         defaults = cls.load_defaults(
             main_default_file_path, environment_default_file_path
         )
+        log.debug(
```
Comment (disrupted) [inline kpops/components/base_components/base_defaults_component.py:131]: why did you move the log position? afaict it was correct before

File: kpops/utils/yaml.py [hunk 2]
```diff
@@ -24,6 +27,9 @@ def load_yaml_file(
     file_path: Path, *, substitution: Mapping[str, Any] | None = None
 ) -> dict | list[dict]:
     with file_path.open() as yaml_file:
+        log.debug(
+            f"Picked up: {file_path.resolve().relative_to(file_path.resolve().parents[1])}"
```
Comment (disrupted) [inline kpops/utils/yaml.py:31]: I have a question about the relative path. What happens if the pipeline folder is nested deeper? e.g. `pipelines/foo/bar/.../baz/pipeline.yaml`
would it only print `baz/pipeline.yaml`? imo relative to cwd as we normally do makes the most sense

File: tests/pipeline/test_generate.py [hunk 3]
```diff
@@ -265,6 +265,28 @@ def test_with_env_defaults(self, snapshot: SnapshotTest):
         enriched_pipeline: list = yaml.safe_load(result.stdout)
         snapshot.assert_match(enriched_pipeline, "test-pipeline")
 
+    def test_with_env_all_resources(self, snapshot: SnapshotTest):
```
Comment (disrupted) [inline tests/pipeline/test_generate.py:268]: what is this testing? (sorry if that's a stupid question, but I didn't get it)

Comment (raminqaf) [reply tests/pipeline/test_generate.py:268]: _There is no stupid questions only stupid answers_ 🧘‍♂️

It is just a mock test I added to check the debug logs. The folder contains all possible files with their envs (config.yaml, config_dev.yaml, pipeline.yaml,...)
It was easier for me to visualise the pickup file logs. This would be removed and later on I will add log captors for the picked up files!

File: kpops/components/base_components/base_defaults_component.py (hunk 1 above)
Comment (raminqaf) [reply kpops/components/base_components/base_defaults_component.py:192]: I moved it because it was before the method `get_defaults_file_paths` which returns the default path and does not do any "enriching". The enriching is happening in the `update_nested_pair`. Therefore I moved it right before the method call

File: kpops/components/base_components/base_defaults_component.py [hunk 4]
```diff
@@ -238,7 +238,7 @@ def defaults_from_yaml(path: Path, key: str) -> dict:
     if value is None:
         return {}
     log.debug(
-        f"\tFound defaults for component type {typer.style(key, bold=True, fg=typer.colors.MAGENTA)} in file:  {path}"
+        f"Found defaults for component type {typer.style(key, bold=True, fg=typer.colors.MAGENTA)}"
```
Comment (disrupted) [inline kpops/components/base_components/base_defaults_component.py:241]: Forgot that we have this 😄 isn't that even better than printing all defaults.yaml in the enrichment path, because this only prints if it actually finds something for a specific component?

File: kpops/cli/registry.py [hunk 5]
```diff
@@ -53,11 +55,18 @@ def find_class(module_name: str, baseclass: type[T]) -> type[T]:
 
 def _find_classes(module_name: str, baseclass: type[T]) -> Iterator[type[T]]:
     module = importlib.import_module(module_name)
+    if module.__file__ and not module_name.startswith(KPOPS_MODULE):
+        file_path = Path(module.__file__)
+        log.debug(f"Picked up: {file_path.resolve().relative_to(Path.cwd())}")
```
Comment (disrupted) [inline kpops/cli/registry.py:60]: tbh, I'd prefer printing the filepath as-is, without resolving. If there a symbolic links and the like, as a user I would wanna see the link, not the resolved path

Comment (raminqaf) [reply kpops/cli/registry.py:60]: tbh, I prefer to have it like that. Basically we are showing the files paths from the `pipeline_base_dir` instead of the root. The parent path of the `pipeline_base_dir` is less useful for the user. With all the changes, the logs would look like this:

```console
Yaml - Picked up: config.yaml
Registry - Picked up: tests/pipeline/test_components/__init__.py
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/pipeline.yaml
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/defaults_development.yaml
Yaml - Picked up: tests/pipeline/resources/pipeline-with-env-defaults/defaults.yaml
BaseDefaultsComponent - Found defaults for component type streams-app in defaults.yaml
Yaml - Picked up: tests/pipeline/resources/defaults.yaml
BaseDefaultsComponent - Found defaults for component type streams-app in defaults.yaml
Yaml - Picked up: tests/pipeline/defaults.yaml
BaseDefaultsComponent - Found defaults for component type kafka-app in defaults.yaml
BaseDefaultsComponent - Found defaults for component type kafka-app in defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in defaults_development.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in defaults.yaml
BaseDefaultsComponent - Found defaults for component type kubernetes-app in defaults.yaml
BaseDefaultsComponent - Found defaults for component type pipeline-component in defaults.yaml
BaseDefaultsComponent - Enriching component of type streams-app
```

File: kpops/components/base_components/base_defaults_component.py (hunk 4 above)
Comment (raminqaf) [reply kpops/components/base_components/base_defaults_component.py]: I would still keep the log in the yaml.py since, in that place, all the pickup YAMLs are logged. For clarity, I added the default paths relative to the cwd here.

File: kpops/cli/registry.py (hunk 5 above)
Comment (disrupted) [reply kpops/cli/registry.py:60]: but why do we have to resolve anything at all? isn't the path already relative an what we want?

Comment (raminqaf) [reply kpops/cli/registry.py:60]: Good point! in the registry we don't really need to resolve because we are getting the `Path(module.__file)`.

File: kpops/components/base_components/base_defaults_component.py [hunk 6]
```diff
@@ -238,8 +238,9 @@ def defaults_from_yaml(path: Path, key: str) -> dict:
     value = content.get(key)
     if value is None:
         return {}
+    default_path = path.resolve().relative_to(Path.cwd())
```
Comment (disrupted) [inline kpops/components/base_components/base_defaults_component.py:241]: there's no need for `resolve()`, path is already absolute. also, could you move this to the log statement

**Review by disrupted — APPROVED** [review]
